src/libPersistence/BlockStorage.cpp

```diff
@@ -425,6 +500,8 @@ bool BlockStorage::DeleteTxBody(const dev::h256& key) {
 bool BlockStorage::GetAllDSBlocks(std::list<DSBlockSharedPtr>& blocks) {
   LOG_MARKER();
 
+  shared_lock<shared_timed_mutex> g(m_mutexDsBlockchain);
+
   leveldb::Iterator* it =
       m_dsBlockchainDB->GetDB()->NewIterator(leveldb::ReadOptions());
```

Comment:
Maybe let `m_mutexDsBlockchain` to protect `leveldb::Iterator* it = m_dsBlockchainDB->GetDB()->NewIterator(leveldb::ReadOptions());` only

Reply:
cannot do that...the iterator is keep changing in the for loop